### PR TITLE
NAS-109694 / 12.0 / Initial protocol testing for SMB using pylibsmb

### PR DESCRIPTION
**Pull request by anodos325**
Pylibsmb gives more fine-grained control over the SMB operations
performed on the server. Initial commit sets framework for doing these
additional tests as well as implements setting DOS modes through SMB1/2 CREATE.

### CODE REVIEW

**Comment by anodos325** [comment]
Note: this change will require net/samba412 (or some recent samba) to be installed on the server performing the tests
Minimal smb.conf (or smb4.conf) on the API testing server will be:
```
[global]
client min protocol = NT1
```
To allow python binding to perform SMB1 connections.

**Comment by anodos325** [comment]
Future commits will expand functionality for SMB1/2 tests to include shadow copies and alternate data streams.

**Review by william-gr — APPROVED** [review]

File: tests/api2/smb_protocol.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python3
+
+# Author: Eric Turgeon
```
Comment (ericbsd) [inline tests/api2/smb_protocol.py:3]: Should remove this or change the name I did not write the code.

**Review by ericbsd — CHANGES_REQUESTED** [review]

**Review by ericbsd — APPROVED** [review]
LGTM

**Pull request merged**